### PR TITLE
deactivate apisec tests for sampling

### PR DESCRIPTION
**Pull request by christophe-papazian**
## Motivation

Sampling mechanism is currently changing. Previous RFC is deprecated and a new RFC is soon to be approved.

## Changes

deactivate irrelevant test.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

**Review by robertomonteromiguel — APPROVED** [review]
From the point of view of CI and system-tests framework, it's right

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
thinking about it, why don't you use the irrelevant method decorator, specifying the version from which these tests would be irrelevant?

**Review by robertomonteromiguel — APPROVED** [review]

**Comment by christophe-papazian** [comment]
@simon-id 
For the tests, it's breaking. Otherwise, in a production environment, backend should see more schemas and tracers should compute less schemas. It's a win-win.

**Comment by simon-id** [comment]
Where is the new test btw ?

**Comment by christophe-papazian** [comment]
@simon-id Yet to be done.

**Comment by simon-id** [comment]
Then there is no point in removing coverage until the new test and new implementation is rolled out.

**Comment by cbeauchesne** [comment]
What about using the legacy trick ? You don't have to touch manifest, and you keep testing libs that have not yet implemented the new behavior

```python
class TestSampling:
    @irrelevant(context.library > "python@1.2", reason="Python implemented the new behavior at v1.2")
    def test_legacy_behavior(self):
         # old stuff

    @irrelevant(context.library <= "python@1.2", reason="Python implemented the new behavior at v1.2")
    @missing_feature(library="java", reason="java still use the old behavior")
    def test_behavior(self):
         # new stuff
```

**Comment by simon-id** [comment]
@christophe-papazian Disable it for your own library if you want, but I want to keep the coverage on mine, until we implement the new feature and new test, thank you

File: tests/appsec/api_security/test_apisec_sampling.py [hunk 1]
```diff
@@ -37,6 +38,7 @@ def setup_sampling_rate(self):
             for _ in range(self.N ** 2)
         ]
 
+    @irrelevant(True, reason="RFC is deprecated by a newer RFC. New tests will be implemented")
```
Comment (simon-id) [inline tests/appsec/api_security/test_apisec_sampling.py:41]: @christophe-papazian i see you've made a commit `do not touch other libs` but doesn't this affect other libs ?

**Pull request merged**